src/abb_system_position_only.cpp

```diff
@@ -14,264 +14,259 @@
-				  RCLCPP_WARN(rclcpp::get_logger("ABBSystemPositionOnlyHardware"),
-						"configure RAPID execution state is UNDEFINED (might happen first time after controller start/restart). Try to restart the RAPID program.");
-			}else{
-				wait = egm_interface_->getStatus().rapid_execution_state() != abb::egm::wrapper::Status_RAPIDExecutionState_RAPID_RUNNING;
-			}
-		}else{
-			  RCLCPP_INFO(rclcpp::get_logger("ABBSystemPositionOnlyHardware"), "Not Connected to Robot...");
-		}
-		rclcpp::sleep_for(500ms);
-	}
-	if(wait){  // if still not in the right state, exit
-		  RCLCPP_FATAL(rclcpp::get_logger("ABBSystemPositionOnlyHardware"), "Could NOT Connect to Robot");
-		//exit(EXIT_FAILURE); // HACK FOR NOW
-		return CallbackReturn::ERROR; // TODO: this does not seem to stop anything, and the code continues to read/write
-	}
-	  RCLCPP_INFO(rclcpp::get_logger("ABBSystemPositionOnlyHardware"), "CONNECTED!");
-
-
-	//SET INITIAL STATE TO CURRENT ANGLES, error if something isn't right
-	if(!egm_interface_->waitForMessage(500)){ //in [ms]
-		RCLCPP_FATAL(rclcpp::get_logger("ABBSystemPositionOnlyHardware"), "NO Message Received");
-		//exit(EXIT_FAILURE); // HACK FOR NOW
-		return CallbackReturn::ERROR; // TODO: this does not seem to stop anything, and the code continues to read/write
-	}
-
-	egm_interface_->read(&input_);
-	sequence_number_ = input_.header().sequence_number();
-
-	if(sequence_number_ <= 0){
-		RCLCPP_FATAL(rclcpp::get_logger("ABBSystemPositionOnlyHardware"), "Sequence number < 0");
-		//exit(EXIT_FAILURE); // HACK FOR NOW
-		return CallbackReturn::ERROR; // TODO: this does not seem to stop anything, and the code continues to read/write
-	}
-
-	output_.Clear();
-	output_pos_.Clear();
-	current_positions_.CopyFrom(input_.feedback().robot().joints().position());
-
-	if((size_t)current_positions_.values_size() != hw_states_.size()){
-		RCLCPP_FATAL(rclcpp::get_logger("ABBSystemPositionOnlyHardware"),
-				"Robot does not match expected number of joints");
-		//exit(EXIT_FAILURE); // HACK FOR NOW
-		return CallbackReturn::ERROR; // TODO: this does not seem to stop anything, and the code continues to read/write
-	}
-
-	output_.mutable_robot()->mutable_joints()->mutable_position()->CopyFrom(current_positions_);
-	output_pos_.mutable_robot()->mutable_joints()->mutable_position()->CopyFrom(current_positions_);
-
-	for(size_t i=0; i<hw_states_.size(); i++){
-		hw_states_[i] = current_positions_.values(i)/180.0*3.14159;
-		hw_commands_[i] = current_positions_.values(i)/180.0*3.14159; //QUESTION: should I do this here?
-	}
-
-
-	RCLCPP_INFO(rclcpp::get_logger("ABBSystemPositionOnlyHardware"),
-	"System Sucessfully started!");
-
-	return CallbackReturn::SUCCESS;
+  RCLCPP_INFO(rclcpp::get_logger("ABBSystemPositionOnlyHardware"), "on_activate()");
+
+  thread_group_.create_thread(boost::bind(&boost::asio::io_service::run, &io_service_));
+
+  bool wait = true;
+  int num_tries = 100;
```

Comment:
Let's move this to a constant in an anonymous namespace at the top of the file and rename it:

`size_t const NUM_CONNECTION_TRIES = 0;`

Reply:
This is the last remaining change. Do you want to do it in the other PR or here?